File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPSaturation.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPSaturation implements PlayerStateProperty<Float> {
    @Override
    public String getConfigName() {
        return "saturation";
    }

    @Override
    public Float getDefaultSerializedProperty() {
        return 20F;
    }

    @Override
    public Float getSerializedPropertyFromPlayer(Player player) {
        return player.getSaturation();
    }

    @Override
    public void restoreProperty(Player player, Float saturation) {
        player.setSaturation(saturation);
    }

    @Override
    public Float getValueFromConfig(ConfigurationSection config) {
        return ((Double) config.getDouble(getConfigName())).floatValue();
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPSaturation.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPSaturation extends PlayerStateProperty[Float]:
  override def getConfigName: String = "saturation"

  override def getDefaultSerializedProperty: Float = 20

  override def getSerializedPropertyFromPlayer(player: Player): Float =
    player.getSaturation

  override def restoreProperty(player: Player, saturation: Float): Unit =
    player.setSaturation(saturation)

  override def getValueFromConfig(config: ConfigurationSection): Float =
    config.getDouble(getConfigName).toFloat